core/commonMain/src/Arguments.kt

```diff
@@ -9,17 +9,32 @@ import kotlin.reflect.KProperty
 internal data class CLIEntityWrapper(var entity: CLIEntity<*>? = null)
 
 /**
- * The base class for a command line argument or an option.
+ * Base interface for all possible types of entities with default anf required values.
```

Comment:
```suggestion
 * Base interface for all possible types of entities with default and required values.
```

Reply:
Overall, this wording seems to be misleading: it can be perceived that `DefaultRequiredType` is implemented by an option/argument itself or by their subtypes, rather than by the type of their marker type argument. We need to document this distinction more clearly, perhaps at the use sites of these marker types.